### PR TITLE
Adjust Itinerary View Scroll Behavior

### PR DESCRIPTION
**Pull request by miles-grant-ibigroup**
At the moment, the itineraries scroll in their own box while the batch planner stays static. This PR modifies this so the _entire column_ scrolls. This can potentially be changed so that the entire column scrolls only when the container is small (due to something like high zoom).

As part of this work the layout of the batch routing panel is modified. The previous one didn't allow itself to be scrolled. 

The new behavior only kicks in at high zoom levels.

### CODE REVIEW

File: lib/components/form/form.css [hunk 1]
```diff
@@ -414,3 +414,26 @@
 .otp .date-time-modal .date-time-selector.with-desktop-touchscreen input::-webkit-datetime-edit {
   margin-inline-end: 8px;
 }
+
+.batch-routing-panel-location-fields {
+  display: grid;
+  grid-template-rows: 1fr 1fr;
+  grid-template-columns: 1fr 40px 10px;
```
Comment (binh-dam-ibigroup) [inline lib/components/form/form.css:421]: The grid stuff that follows is not needed if you do for this CSS class:
```suggestion
  display: block;
  position: relative;
```

Note that the switch button container's existing CSS should handle the positioning correctly already with this suggestion.

File: lib/components/app/batch-routing-panel.js [hunk 2]
```diff
@@ -68,8 +71,7 @@ class BatchRoutingPanel extends Component {
         <div
           className="desktop-narrative-container"
           style={{
-            flexGrow: 1,
-            overflowY: 'hidden'
+            flexGrow: 1
           }}
```
Comment (binh-dam-ibigroup) [inline lib/components/app/batch-routing-panel.js:75]: Remove all of it. (allowing scrolling removes any bounds that would apply for a flex layout)

Comment (binh-dam-ibigroup) [reply lib/components/app/batch-routing-panel.js:75]: Also, around Line 33, remove `display: 'flex'` and `flexDirection` from the `ViewerContainer` style.

**Review by binh-dam-ibigroup — CHANGES_REQUESTED** [review]
The layout can be simplified (the grid for location fields is a bit overkill) and styles cleaned up.

File: lib/components/form/form.css (hunk 1 above)
Comment (miles-grant-ibigroup) [reply lib/components/form/form.css:421]: Screenshot 2023-02-08 at 1 21 18 PM
Not sure if I missed something but this doesn't exactly work

File: lib/components/app/batch-routing-panel.js (hunk 2 above)
Comment (miles-grant-ibigroup) [reply lib/components/app/batch-routing-panel.js:75]: Removing this causes the background of the pane to not extend to the full window height.

File: lib/components/app/responsive-webapp.js [hunk 3]
```diff
@@ -212,7 +212,7 @@ class ResponsiveWebapp extends Component {
             <Col className="sidebar" md={4} sm={6}>
               {/* Note: the main tag provides a way for users of screen readers to skip to the
                   primary page content (tabindex = -1 needed for programmatic navigation skip). */}
-              <main tabIndex={-1}>
+              <main style={{ overflow: 'scroll' }} tabIndex={-1}>
```
[the remark this answers is not in the document]
Comment (miles-grant-ibigroup) [reply lib/components/app/responsive-webapp.js]: Good call thank you!

File: lib/components/form/form.css (hunk 1 above)
Comment (binh-dam-ibigroup) [reply lib/components/form/form.css:421]: For the switch button container, apply `position: absolute; right: ..., top: ...` and for its parent element, apply `position: relative`.

Comment (miles-grant-ibigroup) [reply lib/components/form/form.css:421]: I tried this and couldn't get it to work, since the container moves around so much (and forget about mobile). Maybe you'll have more luck

**Comment by binh-dam-ibigroup** [comment]
Note that the change to scroll the location fields creates this side effect where the location field popup list gets clipped by the `<main>` element (see screenshot).

image

**Comment by miles-grant-ibigroup** [comment]
@binh-dam-ibigroup rules are refactored to only apply at high zoom

File: lib/components/app/app.css [hunk 4]
```diff
@@ -102,3 +102,23 @@
...
 }
+
+/** These changes kick in when the map is hidden. These rules ensure that the entire "sidebar"
+    (which at this point fills the entire screen) scrolls as a single unit.
+    These rules can only apply at this high zoom level because the location field dropdown is
+    otherwise "stuck" inside the sidebar and can't overlap the map. */
+@media (max-width: 768px) {
+  .container-fluid > .navbar-header {
+    margin-bottom: -7px;
+  }
```
Comment (binh-dam-ibigroup) [inline lib/components/app/app.css:113]: Revert this change, it causes the mobile title to be not vertically centered.

**Review by binh-dam-ibigroup — APPROVED** [review]
One thing to revert on the mobile title bar, please, but the scrolling threshold is clever.

**Review by daniel-heppner-ibigroup — APPROVED** [review]
